src/server/standalone.js

```diff
@@ -194,9 +195,19 @@ function StandAloneServer(gmeConfig) {
             }
         });
 
-        __httpServer.listen(gmeConfig.server.port, function () {
+        // something link
+        // https://github.com/rubenv/node-autoquit
+        // should be considered to shut the service down if it
+        // is not getting much use.
+
+        // https://github.com/herzi/systemd-socket
+        // https://nodejs.org/api/http.html#http_server_listen_handle_callback
+        __httpServer.listen(systemdSocket(0) || gmeConfig.server.port, function () {
```

Comment:
It looks like `systemdSocket(0)` returns the port number... If so, why don't you just add this to your config files and set `gmeConfig.server.port` to this value?